plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/shipping-calculator/address.tsx

```diff
@@ -34,16 +37,32 @@ const ShippingCalculatorAddress = ( {
 		( select ) => {
 			return {
 				hasValidationErrors:
-					select( validationStore ).hasValidationErrors,
+					select( validationStore ).hasValidationErrors(),
 				isCustomerDataUpdating:
 					select( CART_STORE_KEY ).isCustomerDataUpdating(),
 			};
 		}
 	);
+	const { defaultFields } = useCheckoutAddress();
+	const formFields = useFormFields(
+		addressFields,
+		defaultFields,
+		'shipping',
+		address.country
+	);
+
+	const hasRequiredFields = () => {
+		const requiredFields = formFields
+			.filter( ( field ) => field.required && ! field.hidden )
+			.map( ( field ) => field.key );
+		return requiredFields.every(
+			( field ) => address[ field ] && address[ field ].trim() !== ''
+		);
+	};
```

Comment:
It's bad practice to create a closure over some value from a hook in case the callback could run with a stale value before the next re-render.

This could be converted to useCallback to ensure the callback always has the latest value:

```suggestion
	const hasRequiredFields = useEffect(() => {
		const requiredFields = formFields
			.filter( ( field ) => field.required && ! field.hidden )
			.map( ( field ) => field.key );
		return requiredFields.every(
			( field ) => address[ field ] && address[ field ].trim() !== ''
		);
	}, [ formFields ] );
```

Reply:
Thanks for pointing this out. With https://github.com/woocommerce/woocommerce/pull/55853/commits/37af800851e0b4673ced601cc3dbfc4fadb35b87, I've wrapped this call in a `useCallback()` and improved the readability.